### PR TITLE
Moved get_basket function

### PR DESCRIPTION
**Pull request by clintonb**
Moved the get_basket function from api/data to a class method on the Basket model.

ECOM-2583

### CODE REVIEW

**Comment by clintonb** [comment]
FYI @edx/ecommerce

File: ecommerce/extensions/basket/models.py [hunk 1]
```diff
@@ -14,6 +15,28 @@ class Basket(AbstractBasket):
...
         return OrderNumberGenerator().order_number(self)
 
+    @classmethod
+    def get_basket(cls, user):
+        """Retrieve the basket belonging to the indicated user.
+
+        If no such basket exists, create a new one. If multiple such baskets exist,
+        merge them into one.
+        """
+        editable_baskets = Basket.objects.filter(owner=user, status__in=Basket.editable_statuses)
```
Comment (peter-fogg) [inline ecommerce/extensions/basket/models.py]: This and the other lines in this classmethod (and all classmethods) shouldn't mention the class name explicitly; use `cls.objects.filter(...)`.

Comment (clintonb) [inline ecommerce/extensions/basket/models.py]: Good catch. Fixed.

**Comment by peter-fogg** [comment]
:+1:

**Comment by rlucioni** [comment]
:+1:

**Comment by bderusha** [comment]
:+1:

**Pull request merged**